src/modules/hip/kernel/resize.cpp

```diff
@@ -345,6 +345,71 @@ extern "C" __global__ void resize_crop_batch(unsigned char *srcPtr,
     }
 }
 
+extern "C" __global__ void resize_nn_crop_batch(unsigned char *srcPtr,
+                                             unsigned char *dstPtr,
+                                             unsigned int *source_height,
+                                             unsigned int *source_width,
+                                             unsigned int *dest_height,
+                                             unsigned int *dest_width,
+                                             unsigned int *max_source_width,
+                                             unsigned int *max_dest_width,
+                                             unsigned int *xroi_begin,
+                                             unsigned int *xroi_end,
+                                             unsigned int *yroi_begin,
+                                             unsigned int *yroi_end,
+                                             unsigned long long *source_batch_index,
+                                             unsigned long long *dest_batch_index,
+                                             const unsigned int channel,
+                                             unsigned int *source_inc, // use width * height for pln and 1 for pkd
+                                             unsigned int *dest_inc,
+                                             const unsigned int padding,
+                                             const unsigned int type,
+                                             const int in_plnpkdind, // use 1 pln 3 for pkd
+                                             const int out_plnpkdind)
+{
+    int id_x = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    unsigned int src_roi_width = xroi_end[id_z] - xroi_begin[id_z];
+    unsigned int src_roi_height = yroi_end[id_z] - yroi_begin[id_z];
+    float x_ratio = (float)src_roi_width / dest_width[id_z];
+    float y_ratio =(float) src_roi_height / dest_height[id_z];
+    float origin_x = xroi_begin[id_z] + (0.5f * x_ratio);
+    float origin_y = yroi_begin[id_z] + (0.5f * y_ratio);
+
+    int x, y;
+    unsigned long dst_pixIdx = 0;
+
+    if (id_x >= dest_width[id_z] || id_y >= dest_height[id_z])
+    {
+        return;
+    }
+
+    x = (int)(floor(x_ratio * id_x + origin_x));
```

Comment:
I checked, looks like typecasting to int and doing floor operation on a float value yields the same result. But when we have negative numbers the operations would vary. In this case since we will not be getting any negative values I shall retain the operation as typecasting to int